src/util/usertools.c

```diff
@@ -863,17 +863,26 @@ void sss_set_sssd_user_eid(void)
     uid_t uid;
     gid_t gid;
 
+
     if (geteuid() == 0) {
         sss_sssd_user_uid_and_gid(&uid, &gid);
-        seteuid(uid);
-        setegid(gid);
+        if (seteuid(uid) != EOK) {
+            DEBUG(SSSDBG_MINOR_FAILURE, "failed to set euid to %"SPRIuid"\n", uid);
+        }
+        if (setegid(gid) != EOK) {
```

Comment:
In other places in the code for this kind of scenario `errno` is printed and evaluated using `sss_strerror()` (src/providers/ipa/ipa_deskprofile_rules_util.c +1001).

Reply:
This is good point. I added printing errno string to all debug messages.